### ISSUE TITLE
Load React from cdn

### ISSUE THREAD

**Issue by imranweb** [zeit/next.js#7112]
I am trying to load React and React-Dom from cdn.
<Head>

        <script key="vendor-react" crossorigin src="https://unpkg.com/react@16/umd/react.production.min.js"></script>
     <script key="vendor-react-dom" crossorigin src="https://unpkg.com/react-dom@16/umd/react-dom.production.min.js"></script>
      </Head>

How can I ensure these scripts loading before the nextjs bundles (main.js, common.js, index.js, etc)?

**Comment by wouterds** [comment]
Why are you trying to do this? These are included by Next.js by default.

**Comment by imranweb** [comment]
I have monorepo setup where nextjs is being used in multiple packages along
with many third party libraries such as Redux, React-Redux, Redux-Saga,
Styled-Components, etc.
Multiple apps will be deployed and running separately on different port.

Example:
- App1: http://localhost:8081 (home, about, contact)
- App2: http://localhost:8082 (products, product detailed, category, search)
- App3: http://localhost:8083 (checkout, shipping, order)

In above example all three apps will have reverse proxy rules with *nginx*
so if user enter the URL *http://localhost:8080/home
<http://localhost:8080/home>* then it will go to *http://localhost:8081/home
<http://localhost:8081/home>.*
Similarly if user enter the URL *http://localhost:8080/products
<http://localhost:8080/products>* then nginx reverse proxy automatically
redirect to *http://localhost:8082/products
<http://localhost:8082/products>*

I want to exclude the specific third party libraries(mentioned above) from
app's bundles and load it from common resource.
e.g. *http://localhost:8084/vendors.js
<http://localhost:8084/vendors.js> *should
be loaded in all three apps. This vendor contains third party libraries
such as Redux, React-Redux, Redux-Saga, Styled-Components, etc.

The advantage of this vendor approach is user's browse will load the third
party libraries only once and it will be serve from cache after that.

Note: Please see the port in URLs mentioned above.


Thanks
-Imran

**Comment by imranweb** [comment]
Did you get chance to look into it?

**Comment by Timer** [comment]
Hi! Questions like this are better suited for our [Spectrum](https://spectrum.chat/next-js) -- this will likely involve some webpack trickery and a custom `_document`.

Right now, this is not supported OOTB and is currently not on our radar. I'm going to close this as we haven't seen any "+1"s. Thanks!

**Comment by bennidhamma** [comment]
fwiw, we'd be interested in this kind of option.

**Comment by timneutkens** [comment]
Using an external CDN is no longer feasible as browsers no longer share resources between domains.

**Comment by petrovalex** [comment]
+1
I am trying to achieve multiple apps deployments using Multi Zones feature (so gonna be the same domain) and would like to share 3rd party library dependencies.

Do you think it could be achieved with Webpack 5 module federation?
..or webpack.externals?

**Comment by edomonticelli** [comment]
I am also interested.

a section of our site is built with next (search, product catalogues) but others no. I'd like them to have all the same look and feel and share a react UI library. This is included along with react as a umd module in the head.

I spent some time with webpack and so on, but I couldn't get it to work.

**Comment by balazsorban44** [comment]
This issue has been automatically locked due to no recent activity. If you are running into a similar issue, please create a new issue with the steps to reproduce. Thank you.
